ably/rest_channel.go

```diff
@@ -157,6 +160,50 @@ func (c *RESTChannel) PublishMultipleWithOptions(ctx context.Context, messages [
 	return c.PublishMultiple(ctx, messages, options...)
 }
 
+type ChannelDetails struct {
+	ChannelId string        `json:"channelId"`
+	Status    ChannelStatus `json:"status"`
+}
+
+type ChannelStatus struct {
+	IsActive  bool             `json:"isActive"`
+	Occupancy ChannelOccupancy `json:"occupancy"`
+}
+
+type ChannelOccupancy struct {
+	Metrics ChannelMetrics `json:"metrics"`
+}
+
+type ChannelMetrics struct {
+	Connections         int `json:"connections"`
+	PresenceConnections int `json:"presenceConnections"`
+	PresenceMembers     int `json:"presenceMembers"`
+	PresenceSubscribers int `json:"presenceSubscribers"`
+	Publishers          int `json:"publishers"`
+	Subscribers         int `json:"subscribers"`
+}
+
+// Status returns ChannelDetails representing information for a channel
+func (c *RESTChannel) Status(ctx context.Context) (*ChannelDetails, error) {
+	req := &request{
+		Method: "GET",
+		Path:   "/channels/" + c.Name,
+	}
+	res, err := c.client.MakeHTTPRequest(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	defer res.Body.Close()
+	var channelDetails ChannelDetails
+
+	// Decode response body
+	dec := codec.NewDecoder(res.Body, new(codec.MsgpackHandle))
```

Comment:
I think here we just want to call [`decodeResp`](https://github.com/ably/ably-go/blob/main/ably/rest_client.go#L864-L873) which will check the `Content-Type` header to determine whether to decode msgpack or JSON.